Fix demo initialization failure in Marvell board

aggarw13 opened the pull request:
The Marvell board encounteres frequent failures in demo jobs on CI with the following log:

```

13 1588361 [IP-task] vDHCPProcess: offer c0a82344ip

Network connection successful.

14 1888450 [IP-task] vDHCPProcess: offer c0a82344ip

Network connection successful.

15 2188554 [IP-task] vDHCPProcess: offer c0a82344ip

Network connection successful.

16 2488593 [IP-task] vDHCPProcess: offer c0a82344ip

Network connection successful.

17 2693377 [iot_thread] [ERROR][DEMO][2693377] Failed to initialize all the networks configured for18 2693377 [iot_thread] [ERROR][DEMO][2693377] Failed to initialize the demo. exiting...
19 2693377 [iot_thread] [INFO ][DEMO][2693377] -------DEMO FINISHED-------
```

The cause of the failure is from 2 calls made to `WiFi_ConnectAP` in the demo run for Marvell, one from the [aws_demos/application_code/main.c](https://github.com/aws/amazon-freertos/blob/202012.00/vendors/marvell/boards/mw300_rd/aws_demos/application_code/main.c#L436) and the other from the [Network manager](https://github.com/aws/amazon-freertos/blob/202012.00/demos/network_manager/aws_iot_network_manager.c#L479) in the demo framework.

This PR fixes the issue by removing the duplicate call to connect to WiFi network from the Marvell demo `main.c` file.

**Note**: The [`prvWifiConnect`](https://github.com/aws/amazon-freertos/blob/202012.00/vendors/marvell/boards/mw300_rd/aws_demos/application_code/main.c#L363-L524) function has been retained in the `main.c` file for reference logic for WiFi initialization for station and softAP mode in Marvell board for customers.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Commit:
Add logs around WIFI_ConnectAP

Commit:
Remove duplicate Wifi connection logic from main.c in marvell

Commit:
Revert "Add logs around WIFI_ConnectAP"

This reverts commit 7994469f36773b47b76c96005b22ccc1e688c505.

Commit:
Cleanup in Marvell demo main.c from removal of WiFi connection logic

aggarw13 commented:
/bot run checks

xuelix commented:
/bot run checks

xuelix commented:
/bot run checks

xuelix commented:
/bot run checks

xuelix commented:
/bot run checks

xuelix commented:
/bot run checks

xuelix commented:
/bot run checks

ravibhagavandas commented on vendors/marvell/boards/mw300_rd/aws_demos/application_code/main.c:
```diff
@@ -171,13 +155,13 @@ int main( void )
+    /* A simple example to demonstrate key and certificate provisioning in
+     * flash using PKCS#11 interface. This should be replaced
+     * by production ready key provisioning mechanism. */
+    vDevModeKeyProvisioning();
```
Previously the provisioning was done in a startup task, does it make sense to still run this in the startup task?

aggarw13 replied:
Moved it back to the startup task to avoid any stack size issues.

Commit:
Re-instate separate task for running vDevModeKeyProvisioning to make sure there is sufficient stack

Rename function and re-add vTaskDelete for startup task

Revert "Rename function and re-add vTaskDelete for startup task"

This reverts commit cf8beedaacdf7c6568cc2f1f946cb0a71805e9ac.

Fix build

Commit:
Fix build issue
